src/ManagedArray.inl

```diff
@@ -207,6 +207,9 @@ template<typename T>
 template<typename Idx>
 CHAI_INLINE
 CHAI_HOST_DEVICE T& ManagedArray<T>::operator[](const Idx i) const {
+#if defined(CHAI_ARRAY_BOUNDS_CHECK)
+  assert( i>=0 && static_cast<size_t>(i) < m_elems );
```

Comment:
Can you make assert work on the GPU? I have a branch with this functionality, but couldn't get it working.

Reply:
Sorry. I didn't see that. Where is HOST_DEVICE_ASSERT defined? I can't find it.

I actually hadn't tried it on the GPU yet....and I will need to change the test to launch a kernel....but I think there should be support??
https://docs.nvidia.com/cuda/cuda-c-programming-guide/#assertion